src/main/java/me/karakelley/http/controllers/Router.java

```diff
@@ -16,6 +26,17 @@ public Router route(String path, Controller controller) {
 
   @Override
   public Response respond(Request request) {
-    return mappings.getOrDefault(request.getPath(), new InvalidRequestController()).respond(request);
+    if (mappings.containsKey(request.getPath())) {
+      return mappings.get(request.getPath()).respond(request);
+    } else {
+      checkRequestedPath(request);
+      return mappings.getOrDefault(request.getPath(), new InvalidRequestController()).respond(request);
+    }
+  }
+
+  private void checkRequestedPath(Request request) {
+    if (publicDirectory != null && publicDirectory.validResource(request.getPath())) {
+      mappings.put(request.getPath(), new StaticFilesController(publicDirectory, new HtmlGenerator()));
```

Comment:
Why does the router need to know anything about static files controller and html?